scripts/validate-do-not-submit.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+// tslint:disable:no-implicit-dependencies
+import { logging } from '@angular-devkit/core';
+import { execSync } from 'child_process';
+
+
+export interface ValidateCommitsOptions {
+  ci?: boolean;
+  base?: string;
+  head?: string;
+}
+
+
+export default async function (argv: ValidateCommitsOptions, logger: logging.Logger) {
+  logger.info('Getting merge base...');
+
+  const prNumber = process.env['CIRCLE_PR_NUMBER'] || '';
+  let baseSha = '';
+  let sha = '';
+
+  if (prNumber) {
+    const url = `https://api.github.com/repos/angular/angular-cli/pulls/${prNumber}`;
```

Comment:
Nice way to determine the base 👍 

We should do something similar in our circleci post checkout step to properly rebase instead of fiddling with the merge commits: https://github.com/angular/angular-cli/blob/e96afb874d4a03a8bf0a9cac9d79dd7dc7db31de/.circleci/config.yml#L31-L48

Reply:
Yeah I'm not sure. We should re-evaluate at some point